action/jsonapi.go

```diff
@@ -107,13 +113,43 @@ func respondMessage(ctx context.Context, s *Action, msgBytes []byte) error {
 	switch message.Type {
 	case "query":
 		return respondQuery(s, msgBytes)
+	case "queryHost":
+		return respondHostQuery(s, msgBytes)
 	case "getLogin":
 		return respondGetLogin(ctx, s, msgBytes)
 	default:
 		return fmt.Errorf("Unknown message of type %s", message.Type)
 	}
 }
 
+func respondHostQuery(s *Action, msgBytes []byte) error {
+	var message queryHostMessage
+	if err := json.Unmarshal(msgBytes, &message); err != nil {
+		return err
+	}
+
+	l, err := s.Store.List(0)
+	if err != nil {
+		return err
+	}
+	choices := make([]string, 0, 10)
+
+	reQuery := fmt.Sprintf("(^|.*/)%s($|/.*)", regexSafeLower(message.Host))
+	if err := searchAndAppendChoices(reQuery, l, &choices); err != nil {
+		return err
+	}
+
+	for len(choices) == 0 && strings.Count(message.Host, ".") > 1 {
```

Comment:
What exactly is this block doing?

Reply:
I refactored that part into a function "isPublicSuffix". I hope it is more clear now. Let me know if it still needs a comment / further refactoring.